docs/migration/index.md

```diff
@@ -0,0 +1,382 @@
+  ibctransfertypes "github.com/cosmos/ibc-go/modules/apps/transfer/types"
+  ibc "github.com/cosmos/ibc-go/modules/core"
+  ibcclient "github.com/cosmos/ibc-go/modules/core/02-client"
+  ibcporttypes "github.com/cosmos/ibc-go/modules/core/05-port/types"
+  ibchost "github.com/cosmos/ibc-go/modules/core/24-host"
+  ibckeeper "github.com/cosmos/ibc-go/modules/core/keeper"
+  
+  // Remove the following packages:
+  // transfer "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer"
+  // ibctransferkeeper "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/keeper"
+  // ibctransfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
+  // ibc "github.com/cosmos/cosmos-sdk/x/ibc/core"
+  // ibcclient "github.com/cosmos/cosmos-sdk/x/ibc/core/02-client"
+  // porttypes "github.com/cosmos/cosmos-sdk/x/ibc/core/05-port/types"
+  // ibchost "github.com/cosmos/cosmos-sdk/x/ibc/core/24-host"
+  // ibckeeper "github.com/cosmos/cosmos-sdk/x/ibc/core/keeper"
+)
+
+var (
+  //...
+  ModuleBasics = module.NewBasicManager(
+    //...
+    slashing.AppModuleBasic{},
+    // Add feegrantmodule.AppModuleBasic{},
+    feegrantmodule.AppModuleBasic{}, // <--
+    ibc.AppModuleBasic{},
+    //...
+  )
+  //...
+)
+
+type App struct {
+  //...
+  // Replace codec.Marshaler with codec.Codec
+  appCodec          codec.Codec // <--
+  // Add FeeGrantKeeper
+  FeeGrantKeeper   feegrantkeeper.Keeper // <--
+}
+
+``go
+func New(...) {
+  //bApp.SetAppVersion(version.Version)
+  bApp.SetVersion(version.Version) // <--
+
+  keys := sdk.NewKVStoreKeys(
+    //...
+    upgradetypes.StoreKey,
+    // Add feegrant.StoreKey
+    feegrant.StoreKey, // <--
+    evidencetypes.StoreKey,
+    //...
+  )
+
+  app.FeeGrantKeeper = feegrantkeeper.NewKeeper(appCodec, keys[feegrant.StoreKey], app.AccountKeeper)  // <--
+  // Add app.BaseApp as the last argument to upgradekeeper.NewKeeper
+  app.UpgradeKeeper = upgradekeeper.NewKeeper(skipUpgradeHeights, keys[upgradetypes.StoreKey], appCodec, homePath, app.BaseApp)
+  
+  app.IBCKeeper = ibckeeper.NewKeeper(
+    // Add app.UpgradeKeeper
+    appCodec, keys[ibchost.StoreKey], app.GetSubspace(ibchost.ModuleName), app.StakingKeeper, app.UpgradeKeeper, scopedIBCKeeper,
+  )
+
+  govRouter.AddRoute(govtypes.RouterKey, govtypes.ProposalHandler).
+    //...
+    // Replace NewClientUpdateProposalHandler with NewClientUpdateProposalHandler
+    AddRoute(ibchost.RouterKey, ibcclient.NewClientUpdateProposalHandler(app.IBCKeeper.ClientKeeper))
+
+  // Replace porttypes with ibcporttypes
+  ibcRouter := ibcporttypes.NewRouter()
+
+  app.mm.SetOrderBeginBlockers(
+    upgradetypes.ModuleName,
+    // Add capabilitytypes.ModuleName,
+    capabilitytypes.ModuleName,
+    minttypes.ModuleName,
+    //...
+    // Add feegrant.ModuleName,
+    feegrant.ModuleName,
+  )
+
+  app.mm.RegisterServices(module.NewConfigurator(
+    // Add app.appCodec
+    app.appCodec,
```

Comment:
Could you add the whole RegisterServices here? 

Does this look right?
```
app.mm.RegisterServices(module.NewConfigurator(
		app.MsgServiceRouter(), 
		app.GRPCQueryRouter(), 
		app.appCodec,
	))
```

Reply:
Done!